fetch.bs

```diff
@@ -3699,132 +3771,143 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
- on <var>response</var>. [[!CSP]]
+    <p class=XXX>Gecko
+    <a href=https://bugzilla.mozilla.org/show_bug.cgi?id=1030660>bug 1030660</a> looks
+    into whether this quirk can be removed.
 
- <li><p>If <var>response</var> is not a
- <a>network error</a> and <var>request</var>'s
- <a for=request>cache mode</a> is not "<code>no-store</code>",
- update <var>response</var> in the HTTP cache for <var>request</var>.
- <!-- XXX xref HTTP cache -->
+   <li><p>Execute
+   <a href=https://w3c.github.io/webappsec-csp/#set-response-csp-list>set <var>response</var>'s CSP list</a>
+   on <var>response</var>. [[!CSP]]
 
- <li>
-  <p>If <i>credentials flag</i> is set and the user agent is not configured to block cookies for
-  <var>request</var> (see <a href=https://tools.ietf.org/html/rfc6265#section-7>section 7</a> of
-  [[!COOKIES]]), then run the "set-cookie-string" parsing algorithm (see
-  <a href=https://tools.ietf.org/html/rfc6265#section-5.2>section 5.2</a> of
-  [[!COOKIES]]) on the <a for=header>value</a>
-  of each <var>header</var> whose <a for=header>name</a> is a <a>byte-case-insensitive</a> match for
-  `<code>Set-Cookie</code>` in <var>response</var>'s <a for=response>header list</a>, if any, and
-  <var>request</var>'s <a for=request>current url</a>.
+   <li><p>If <var>response</var> is not a
+   <a>network error</a> and <var>request</var>'s
+   <a for=request>cache mode</a> is not "<code>no-store</code>",
+   update <var>response</var> in the HTTP cache for <var>request</var>.
+   <!-- XXX xref HTTP cache -->
+
+   <li>
+    <p>If <i>credentials flag</i> is set and the user agent is not configured to block cookies for
+    <var>request</var> (see <a href=https://tools.ietf.org/html/rfc6265#section-7>section 7</a> of
+    [[!COOKIES]]), then run the "set-cookie-string" parsing algorithm (see <a
+    href=https://tools.ietf.org/html/rfc6265#section-5.2>section 5.2</a> of [[!COOKIES]]) on the <a
+    for=header>value</a> of each <var>header</var> whose <a for=header>name</a> is a
+    <a>byte-case-insensitive</a> match for `<code>Set-Cookie</code>` in <var>response</var>'s <a
+    for=response>header list</a>, if any, and <var>request</var>'s <a for=request>current url</a>.
+
+    <p class=note>This is a fingerprinting vector.
+  </ol>
 
-  <p class=note>This is a fingerprinting vector.
+ <li><p>If fetch is <a for=fetch>terminated</a>, providing a <var>reason</var>, set
+ <var>response</var>'s <a for=response>termination reason</a> to <var>reason</var> and return
+ <var>response</var>.
 
  <li>
   <p>Run these substeps <a>in parallel</a>:
 
   <ol>
    <li>
-    <p>Whenever one or more bytes are transmitted from <var>response</var>'s message body, let
-    <var>bytes</var> be the transmitted bytes and run these subsubsteps:
-    <!-- XXX xref message body -->
+    <p>Loop the following steps, breaking immediately if fetch
+    <a for=fetch lt=terminated>terminates</a>, providing a <var>reason</var>:
```

Comment:
"Loop the following steps" -> "While true".

I think we can also drop "immediately" since we always mean immediately (unless we state otherwise).